src/main/kotlin/app/revanced/patches/youtube/layout/returnyoutubedislikes/patch/RYDPatch.kt

```diff
@@ -0,0 +1,80 @@
+package app.revanced.patches.youtube.layout.returnyoutubedislikes.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.impl.BytecodeData
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultError
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Dependencies
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patches.youtube.layout.returnyoutubedislikes.annotations.RYDCompatibility
+import app.revanced.patches.youtube.layout.returnyoutubedislikes.fingerprints.TextComponentSpecParentFingerprint
+import app.revanced.patches.youtube.layout.returnyoutubedislikes.fingerprints.DislikeFingerprint
+import app.revanced.patches.youtube.layout.returnyoutubedislikes.fingerprints.LikeFingerprint
+import app.revanced.patches.youtube.layout.returnyoutubedislikes.fingerprints.RemoveLikeFingerprint
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.videoid.patch.VideoIdPatch
+
+@Patch
+@Dependencies(dependencies = [IntegrationsPatch::class, VideoIdPatch::class])
+@Name("return-youtube-dislikes")
+@Description("Shows the dislike count of videos.")
+@RYDCompatibility
+@Version("0.0.1")
+class RYDPatch : BytecodePatch(
+    listOf(
+        TextComponentSpecParentFingerprint, LikeFingerprint, DislikeFingerprint, RemoveLikeFingerprint
+    )
+) {
+    override fun execute(data: BytecodeData): PatchResult {
+        LikeFingerprint.result!!.mutableMethod.addInstructions(
+            0,
+            """
+            const/4 v0, 1
+            invoke-static {v0}, Lapp/revanced/integrations/patches/ReturnYouTubeDislikesPatch;->sendVote(I)V
+            """
+        )
+        DislikeFingerprint.result!!.mutableMethod.addInstructions(
+            0,
+            """
+            const/4 v0, -1
+            invoke-static {v0}, Lapp/revanced/integrations/patches/ReturnYouTubeDislikesPatch;->sendVote(I)V
+            """
+        )
+        RemoveLikeFingerprint.result!!.mutableMethod.addInstructions(
+            0,
+            """
+            const/4 v0, 0
+            invoke-static {v0}, Lapp/revanced/integrations/patches/ReturnYouTubeDislikesPatch;->sendVote(I)V
+            """
+        )
+
+        VideoIdPatch.injectCall("Lapp/revanced/integrations/patches/ReturnYouTubeDislikesPatch;->newVideoLoaded(Ljava/lang/String;)V")
+
+        val parentResult = TextComponentSpecParentFingerprint.result!!
+        val classDef = parentResult.classDef
```

Comment:
You can use the fingerprints mutable hosting class, using `data.proxy(classDef)` is effectively the same:

https://github.com/revanced/revanced-patcher/blob/21b540418029c0c25baec897c414debbc11ead3a/src/main/kotlin/app/revanced/patcher/fingerprint/method/impl/MethodFingerprint.kt#L59